code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -77,49 +78,87 @@ LINEN BINS
 /obj/item/bedsheet/brown
 	icon = 'icons/obj/bedsheets/bedsheet_brown.dmi'
 
+//////////////////////////////////////////
+// Bedsheet bin
+//////////////////////////////////////////
 /obj/structure/bedsheetbin
-	name = "linen bin"
-	desc = "A linen bin. It looks rather cosy."
-	icon = 'icons/obj/structures/linen_bin.dmi'
-	icon_state = "linenbin-full"
-	anchored = 1
-	var/amount = 20
-	var/list/sheets = list()
-	var/obj/item/hidden = null
+	name                   = "linen bin"
+	desc                   = "A linen bin. It looks rather cosy."
+	icon                   = 'icons/obj/structures/linen_bin.dmi'
+	icon_state             = "linenbin-full"
+	anchored               = TRUE
+	w_class                = ITEM_SIZE_STRUCTURE
+	material               = /decl/material/solid/plastic
+	tool_interaction_flags = TOOL_INTERACTION_ANCHOR | TOOL_INTERACTION_DECONSTRUCT
+	var/stored             = 0  //Currently stored unspawned bedsheets, mainly used by mapped bins
+	var/max_stored         = 20 //Maximum amount of bedsheets that can be put in here
+	var/list/sheets             //Currently spawned bedsheets it contains
+	var/obj/item/hidden         //Object hidden amidst the bedsheets
+
+/obj/structure/bedsheetbin/mapped/Initialize(ml, _mat, _reinf_mat)
+	stored = max_stored //Mapped ones start with some unspawned sheets
+	. = ..()
 
-/obj/structure/bedsheetbin/examine(mob/user)
+/obj/structure/bedsheetbin/dump_contents()
+	//Dump all sheets, even unspawned ones
+	for(var/i = 1 to get_amount())
+		remove_sheet()
 	. = ..()
 
-	if(amount < 1)
+/**Returns the total amount of sheets contained, including unspawned ones. */
+/obj/structure/bedsheetbin/proc/get_amount()
+	return LAZYLEN(sheets)? stored + length(sheets) : stored
```

Comment:
You could just return `stored + length(sheets)` since `+0` is valid.

Reply:
Alright, done!

Reply:
btw why not using LAZYLEN again in TRUE case?

Reply:
LAZYLEN is just a wrapper around length() anyway.